tests/metrics/test_observers.py

```diff
@@ -69,3 +103,121 @@ def test__disability_weight_aggregator():
+        ),
+        "prevalence": lambda _, __: build_table(
+            0.45, year_start - 1, year_end, ["age", "year", "sex", "value"]
+        ),
+    }
+    disability_get_data_funcs_1 = {
+        "disability_weight": lambda _, __: build_table(
+            disability_weight_value_1, year_start - 1, year_end
+        ),
+        "prevalence": lambda _, __: build_table(
+            0.65, year_start - 1, year_end, ["age", "year", "sex", "value"]
+        ),
+    }
+    disability_state_0 = DiseaseState(
+        "sick_cause_0", get_data_functions=disability_get_data_funcs_0
+    )
+    disability_state_1 = DiseaseState(
+        "sick_cause_1", get_data_functions=disability_get_data_funcs_1
+    )
+    model_0 = DiseaseModel(
+        "model_0", initial_state=healthy_0, states=[healthy_0, disability_state_0]
+    )
+    model_1 = DiseaseModel(
+        "model_1", initial_state=healthy_1, states=[healthy_1, disability_state_1]
+    )
+
+    simulation = InteractiveContext(
+        components=[
+            TestPopulation(),
+            model_0,
+            model_1,
+            ResultsStratifier(),
+            DisabilityObserver(),
+        ],
+        configuration=base_config,
+        plugin_configuration=base_plugins,
+    )
+    simulation.step()
+
+    pop = simulation.get_population()
+    sub_pop_mask = {
+        "healthy": (pop["model_0"] == "healthy_0") & (pop["model_1"] == "healthy_1"),
+        "sick_0": (pop["model_0"] == "sick_cause_0") & (pop["model_1"] == "healthy_1"),
+        "sick_1": (pop["model_0"] == "healthy_0") & (pop["model_1"] == "sick_cause_1"),
+        "sick_0_1": (pop["model_0"] == "sick_cause_0") & (pop["model_1"] == "sick_cause_1"),
+    }
+
+    # Population masks values for keys of expected labels in results from the metrics pipeline
+    yld_stratification_mask = {
+        "MEASURE_ylds_due_to_all_causes_SEX_Female": (pop["sex"] == "Female"),
+        "MEASURE_ylds_due_to_all_causes_SEX_Male": (pop["sex"] == "Male"),
+        "MEASURE_ylds_due_to_sick_cause_0_SEX_Female": (pop["model_0"] == "sick_cause_0")
+        & (pop["sex"] == "Female"),
+        "MEASURE_ylds_due_to_sick_cause_0_SEX_Male": (pop["model_0"] == "sick_cause_0")
+        & (pop["sex"] == "Male"),
+        "MEASURE_ylds_due_to_sick_cause_1_SEX_Female": (pop["model_1"] == "sick_cause_1")
+        & (pop["sex"] == "Female"),
+        "MEASURE_ylds_due_to_sick_cause_1_SEX_Male": (pop["model_1"] == "sick_cause_1")
+        & (pop["sex"] == "Male"),
+    }
+
+    # Get pipelines
+    disability_weight = simulation._values.get_value("disability_weight")
+    disability_weight_0 = simulation._values.get_value("sick_cause_0.disability_weight")
+    disability_weight_1 = simulation._values.get_value("sick_cause_1.disability_weight")
+
+    # Check that disability weights are computed as expected
+    for sub_pop_key in ["healthy", "sick_0", "sick_1", "sick_0_1"]:
+        assert np.isclose(
```

Comment:
It would be better to have separate tests for the disability weight pipeline and the YLD outputs of the metrics, but this is fine for now.